(feat) emphasize cody for new users on dotcom

erzhtor opened the pull request:
Closes https://github.com/sourcegraph/sourcegraph/issues/51399.

This PR:
- feat(SearchPageContent.tsx): add CodySignUpCtaSection to dotcom search page for unauthenticated users
- feat(BlobPage.tsx): add TryCodyWidget to dotcom blob page for authenticated users
  - feat(TryCodyWidget.tsx): make it auto-dismissible whenever Cody on web is tried
- feat(CloudSignUpPage.tsx): update feature list and page background
- refactor(GlobalNavbar.tsx): change "About" link text to "About Sourcegraph"
- refactor(SearchPageContent.tsx): update search page content message to reflect new features

## Test plan
- `sg start dotcom`
- Check the search home page
- Sign in and go to any blob page

## Demo

https://user-images.githubusercontent.com/6717049/236134788-642c45c3-8149-4945-b3b6-4c936d229c09.mov


<!-- All pull requests REQUIRE a test plan: https://docs.sourcegraph.com/dev/background-information/testing_principles -->

sourcegraph-buildkite commented:
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.03% (+0.81 kb) | 0.66% (+97.40 kb) 🔺 | 0.81% (+96.58 kb) 🔺 | 0.13% (+1) 🔺 |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/emphasize-cody-for-new-users-on-dotcom/compare-report.html;tab=live_object) for a full comparison between the commits 4daba1a and af29d8e or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

sashaostrikov reviewed (approved):
Nice, thanks for demo!

erzhtor commented:
@rrhyne updated PR discussed on the call:
- added signed in the version of CTA section
- added event logs on actions & links
- removed the old "try Cody experimental" CTA section
- other minor UI fixes

Here is a recording of the latest state:

https://user-images.githubusercontent.com/6717049/236298100-1d633f51-41f4-4c01-aeeb-09f6a32c78d8.mov

rrhyne commented:
One QA change for now:

* [Change of header size](https://www.figma.com/file/DIk3eGdgKszigQVec7Y8SD/Frictionless-cody?type=design&node-id=278-25037&t=GRcC9SGwmUYjKWcK-4)

One feature request that I hope isn't too big for tomorrow... marketing/ops needs us to add a form after users complete signup that helps us capture leads. Details here:
* https://sourcegraph.slack.com/archives/C055X672BV0/p1683233079857649?thread_ts=1683221390.070339&cid=C055X672BV0

thenamankumar commented on client/web/src/auth/components/ExternalsAuth.module.scss:
```diff
@@ -0,0 +1,36 @@
+    border: none;
+    width: 100%;
+    padding: 0.75rem 1rem;
+    border-radius: 4px;
```
the linter will demand rem.

thenamankumar replied:
0.25rem

erzhtor replied:
This was an existing style, I just moved it to a separate file with its component. Also, I don't think that we use rem for borders form what I see from existing codebase.

thenamankumar commented on client/web/src/auth/components/FeatureList.tsx:
```diff
@@ -0,0 +1,25 @@
+    <ul className="list-unstyled">{children}</ul>
+)
+
+export const FeatureList = Object.assign(List, { Item })
```
what?

erzhtor replied:
This is to make it work as a namespaced component, e.g., `<FeatureList.Item ... />`

thenamankumar commented on client/web/src/storm/pages/SearchPage/TryCodyCtaSection.tsx:
```diff
@@ -0,0 +1,159 @@
+
+import styles from './TryCodyCtaSection.module.scss'
+
+const MeetCodySVG: React.FC = () => (
```
Move this SVG to a separate file.

erzhtor replied:
https://github.com/sourcegraph/sourcegraph/pull/51451#discussion_r1186009747

thenamankumar commented on client/web/src/storm/pages/SearchPage/TryCodySignUpCtaSection.tsx:
```diff
@@ -0,0 +1,161 @@
+
+import styles from './TryCodySignUpCtaSection.module.scss'
+
+const MeetCodySVG: React.FC = () => (
```
Move the SVG to a separate file.

erzhtor replied:
I want to keep this as is for now, as I intentionally placed it inside the file, as the component codebase is relatively simple and concise, and also SVG is not reused anywhere else. I think it would make sense to move to a separate file when component complexity and code start to grow, to simplify it. Let me know if you think otherwise.

thenamankumar commented on client/web/src/storm/pages/SearchPage/TryCodySignUpCtaSection.tsx:
```diff
@@ -0,0 +1,161 @@
+    className,
+    telemetryService,
+}) => {
+    const onSignUpClick = useCallback(
```
Extra render of a single button element is not worth use of useCallback.
